src/consts.rs

```diff
@@ -311,13 +332,32 @@ impl<'c, 'cc> ConstEvalContext<'c, 'cc> {
                     UnNeg => constant_negate(o),
                     UnUniq | UnDeref => Some(o),
                 }),
-            ExprBinary(op, ref left, ref right) =>
-                self.binop(op, left, right),
+            ExprBinary(op, ref left, ref right) => self.binop(op, left, right),
+            ExprCast(ref value, ref ty) => self.cast(value, ty),
+            ExprIndex(ref x, ref index) =>
+                if let ExprVec(ref v) = x.node {
```

Comment:
True. On the other hand, if `x` is a constant that resolves to a vector (well, array actually), we don't catch that here.